auth/reset-password.php

```diff
@@ -0,0 +1,125 @@
+<?php
+require("../template/top.php");
+require("../api/discord/bots/admin.php");
+if (isset($_GET['token'])) {
+    do {
+        $token = $_GET['token'];
+
+        $q = $db->query("SELECT * FROM `password_reset_tokens` WHERE
+                                            `token` = '" . $db->real_escape_string($token) . "' AND
+                                            `expires` >= NOW()
+                                             LIMIT 1");
+        if ($q->num_rows == 0) {
+            $error = "This password reset link is not valid or has already expired.";
+            break;
+        } else if (isset($_POST['email'])) {
+            $r = $q->fetch_array(MYSQLI_ASSOC);
+            $q = $db->query("SELECT * FROM users WHERE id = '" . $db->real_escape_string($r['uid']) . "' LIMIT 1");
+
+            $email = $_POST['email'];
+            $password1 = $_POST['password1'];
+            $password2 = $_POST['password2'];
+
+            do {
+                if ($q->num_rows != 1) {
+                    $error = "Failed to find the user associated with this reset request.";
+                    break;
+                } else {
+                    $r = $q->fetch_array(MYSQLI_ASSOC);
+                }
+
+                if (empty($password1)) {
+                    $error = "You must enter a password";
+                    break;
+                } else if ($password1 !== $password2) {
+                    $error = "The passwords you entered do not match";
+                    break;
+                } else if ($email != $r['email']) {
+                    $error = "The e-mail address you entered is not valid for this reset link";
+                    break;
+                } else {
+                    $hash = password_hash($password1, PASSWORD_BCRYPT, array('cost' => 12));

```

Comment:
I believe this can return false on failure to hash with the BCrypt algo

https://www.php.net/manual/en/function.password-hash.php